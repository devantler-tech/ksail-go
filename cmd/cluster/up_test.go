package cluster //nolint:testpackage // Access internal helpers without exporting them.

import (
	"testing"

	"github.com/devantler-tech/ksail-go/cmd/cluster/testutils"
)

// TestHandleUpRunE exercises success and validation error paths.

func TestHandleUpRunE(t *testing.T) { //nolint:paralleltest
	t.Run("success", func(t *testing.T) { //nolint:paralleltest
		cmd, manager, output := testutils.NewCommandAndManager(t, "up")
		testutils.SeedValidClusterConfig(manager)

		err := HandleUpRunE(cmd, manager, nil)
		if err != nil {
			t.Fatalf("expected no error, got %v", err)
		}

		assertOutputContains(t, output.String(), "Cluster created and started successfully")
	})

	t.Run("validation error", func(t *testing.T) { //nolint:paralleltest // uses t.Chdir
		testutils.RunValidationErrorTest(t, "up", HandleUpRunE)
	})
}
