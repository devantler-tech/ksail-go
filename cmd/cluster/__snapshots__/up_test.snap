[TestHandleUpRunE/success - 1]
⏳ Load config...
► loading ksail config
► using default config
✔ config loaded [stage: *]
🚀 Create cluster...
► creating cluster
✔ cluster created [stage: *|total: *]

---
